scripts/pin-to-cluster.sh

```diff
@@ -12,17 +12,23 @@ if [[ $# -eq 0 ]] ; then
   echo 'CLUSTER_USER="who" \'
   echo 'CLUSTER_PASSWORD="_secret_" \'
   echo 'GITHUB_TOKEN="_secret" \'
-  echo 'CIRCLE_SHA1="bf3aae3bc98666fbf459b03ab2d87a97505bfab0" \'
-  echo 'CIRCLE_PROJECT_USERNAME="ipfs-shipyard" \'
-  echo 'CIRCLE_PROJECT_REPONAME="ipld-explorer" \'
+  echo 'SHA1="bf3aae3bc98666fbf459b03ab2d87a97505bfab0" \'
+  echo 'PROJECT_USERNAME="ipfs-shipyard" \'
+  echo 'PROJECT_REPONAME="ipld-explorer" \'
   echo './pin-to-cluster.sh <pin name> <input root dir to pin recursivly>'
   exit 1
 fi
 
 HOST=${CLUSTER_HOST:-"/dnsaddr/cluster.ipfs.io"}
+# Default to CIRCLE_CI variables when not set
+: "${SHA1:=$CIRCLE_SHA1}"
+: "${PROJECT_USERNAME:=$CIRCLE_PROJECT_USERNAME}"
+: "${PROJECT_REPONAME:=$CIRCLE_PROJECT_REPONAME}"
```

Comment:
pls enlightne me. What does the preeceeding colon do here?

Reply:
<img width="1211" alt="image" src="https://user-images.githubusercontent.com/356110/173560606-69a78e32-836f-46ff-997b-e4279c8d7af5.png">


[Source](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) ( grep for `The following table` )

Reply:
Ah sorry I misunderstood - yes, this is correct. If you don't add the `:` it will try to "run" the result.

Reply:
Man, I copypasted from SO

Reply:
https://stackoverflow.com/a/28085062